src/rewrite_reduce.cpp

```diff
@@ -54,9 +55,59 @@ struct find_softmax
     }
 };
 
+struct find_reduce_mean
+{
+    auto matcher() const { return match::name("reduce_mean"); }
+
+    void apply(module& m, const match::matcher_result& r) const
+    {
+        auto reduce_mean = r.result;
+        auto op          = reduce_mean->get_operator().to_value();
+        auto axes        = op["axes"].to_vector<std::int64_t>();
+        auto input       = reduce_mean->inputs().front();
+
+        bool is_integral = false;
+        double max_n     = 0;
+        input->get_shape().visit_type([&](auto t) {
+            is_integral = t.is_integral();
+            max_n       = t.max();
+        });
+
+        auto dims = input->get_shape().lens();
+        double n  = std::accumulate(
+            axes.begin(), axes.end(), 1, [&](const auto& prod_value, const auto& axis) {
+                auto norm_axis = axis < 0 ? dims.size() + axis : axis;
+                return prod_value * dims.at(norm_axis);
+            });
+
+        // avoid overflow (the larger value will be later handled)
+        if(n >= max_n / 4)
+            return;
```

Comment:
This could still degrade accuracy. 

If some values inside the inputs are closer to `numeric_limits<type>::max()` or `min()` then it can possibly result in inaccurate results. 
e.g. Let's say as the worst case all the  values inside input are `min()` for floating point and `n == max_n / 4` then current logic would underflow to zero i think. v/s in theory it should result in `min()` as the average value. 

Ideally any accumulation should happen in higher precision.  not sure how to best handle it though.